pkgs/development/libraries/wapp/default.nix

```
@@ -0,0 +1,27 @@
+{ lib, fetchurl, tcl }:
+
+tcl.mkTclDerivation {
+  pname = "wapp";
+  version = "unstable-2023-05-05";
+
+  src = fetchurl {
+    url = "https://wapp.tcl-lang.org/home/raw/72d0d081e3e6a4aea91ddf429a85cbdf40f9a32d46cccfe81bb75ee50e6cf9cf?at=wapp.tcldir?ci=9d3368116c59ef16";
+    sha256 = "sha256-poG7dvaiOXMi4oWMQ5t3v7SYEqZLUY/TsWXrTL62xd0=";
+  };
+
+  unpackPhase = ''
+    mkdir -p $out/lib/wapp
+    cp $src $out/lib/wapp/wapp.tcl
+    cat <<EOF > $out/lib/wapp/pkgIndex.tcl
+    package ifneeded wapp 1.0 [list source [file join \$dir wapp.tcl]]
+    EOF
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -p $out/lib/wapp
    cp $src $out/lib/wapp/wapp.tcl
    cat <<EOF > $out/lib/wapp/pkgIndex.tcl
    package ifneeded wapp 1.0 [list source [file join \$dir wapp.tcl]]
    EOF
    
    runHook postInstall
  '';
```

This is not an `unpackPhase`.
Perhaps it should be defined as `installPhase`.